lifecycle-compile.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+mkdir -p /tmp/compile
+mkdir -p /home/vcap/tmp
+mkdir -p /home/vcap/app/.cloudfoundry
+mkdir -p /home/vcap/app/.profile.d
+touch /home/vcap/app/.cloudfoundry/.placeholder
+touch /home/vcap/app/.profile.d/.placeholder
+chown -R vcap:vcap /home/vcap
+
+curl -# -L "https://storage.googleapis.com/golang/go${GO_VERSION}.linux-amd64.tar.gz" | tar -C /tmp/compile -xz
```

Comment:
whoa, I have never seen this kind of `curl` use... is the `#` supposed to be there?

Also, should this script have `set -e` at the beginning so that any failed commands make the script exit with a nonzero exit code?